tests/countries/test_christmas_island.py

```diff
@@ -0,0 +1,241 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.christmas_island import ChristmasIsland, CX, CXR
+from tests.common import CommonCountryTests
+
+
+class TestChristmasIsland(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(2007, 2050)
+        super().setUpClass(ChristmasIsland, years=years, years_non_observed=years)
+        cls.no_estimated_holidays = ChristmasIsland(years=years, islamic_show_estimated=False)
+
+    def test_country_aliases(self):
+        self.assertAliases(ChristmasIsland, CX, CXR)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(ChristmasIsland(years=1984))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(2007, 2050)))
+        obs_dt = (
+            "2011-01-03",
+            "2012-01-02",
+            "2017-01-02",
+            "2022-01-03",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_chinese_new_year(self):
+        name = "Chinese New Year"
+        self.assertHolidayName(
+            name,
+            "2013-02-11",
+            "2016-02-08",
+            "2019-02-05",
+            "2023-01-23",
+        )
+        years_found = {dt.year for dt in self.holidays.get_named(name, lookup="startswith")}
+        self.assertTrue(set(range(2007, 2050)).issubset(years_found))
```

Comment:
```suggestion
        self.assertHolidayName(
            name,
            "2021-02-12",
            "2021-02-13",
            "2022-02-01",
            "2022-02-02",
            "2023-01-23",
            "2023-01-24",
            "2024-02-10",
            "2024-02-11",
            "2025-01-29",
            "2025-01-30",
        )
        self.assertHolidayName(name, range(2007, 2020), range(2021, 2050))
        obs_dt = (
            "2018-02-19",
            "2020-01-28",
            "2020-01-29",
            "2021-02-15",
            "2024-02-12",
            "2024-02-13",
        )
        self.assertHolidayName(f"{name} (observed)", obs_dt)
        self.assertNoNonObservedHoliday(obs_dt)
```
Yes, 2020 turned out to be a bit non-standard, but there is no way to know from which dates the holiday was moved, so we have observed dates only. 🤷‍♂️